core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -528,6 +548,16 @@ class StandaloneDynamicAllocationSuite
     }
   }
 
+  /** Kill the executors on a given host. */
+  private def killExecutorsOnHost(sc: SparkContext, host: String): Boolean = {
+    syncExecutors(sc)
+    sc.schedulerBackend match {
+      case b: CoarseGrainedSchedulerBackend =>
+        b.killExecutorsOnHost(host, false)
```

Comment:
is this test the only reason to expose the `replace` argument?  if so, I think you could change the test to just check for new executors.  `apps.head.executors.keys.toSet` to get the executors.  (you may need to use `keys.toSet`, instead of `.keySet`, because `.keySet` will just be a view which is not static.)